spec/Request/post_controller_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+require 'rails-controller-testing'
+
+RSpec.describe PostsController, type: :controller do
+  describe 'GET #index' do
+    it 'returns a success response' do
+      user = User.create(name: 'Ngoh')
+      user.posts.create(title: 'Post 1')
+      user.posts.create(title: 'Post 2')
+
+      get :index, params: { user_id: user.id }
+
+      expect(response).to have_http_status(200)
+    end
+
+    it 'renders the index template' do
+      user = User.create(name: 'Ngoh')
+
+      get :index, params: { user_id: user.id }
+
+      expect(response).to render_template(:index)
+    end
+
+    it 'assigns the correct user to @user' do
+      user = User.create(name: 'David')
+
+      get :index, params: { user_id: user.id }
+
+      expect(assigns(:user)).to eq(user)
+    end
+
+    it 'assigns the user posts to @posts' do
+      user = User.create(name: 'David')
+      post1 = user.posts.create(title: 'Post 1')
+      post2 = user.posts.create(title: 'Post 2')
+
+      get :index, params: { user_id: user.id }
+
+      expect(assigns(:posts)).to match_array([post1, post2])
+    end
+  end
+
+  describe 'GET #show' do
+    it 'returns a success response' do
+      post = Post.create(title: 'Post 1')
+
+      get :show, params: { id: post.id }
+
+      expect(response).to have_http_status(200)
+    end
```

Comment:
Well! you've made good progress, but there's a test case that is currently failing:

- [ ] **Fix Failing Test Case:** The test case is encountering an error related to routing: `No route matches {:action=>"show", :controller=>"posts", :id=>nil}`. To resolve this issue, you may need to revisit your test code and ensure that the necessary data, such as the `:id` parameter, is provided or handled correctly.
![image](https://github.com/NalianAchei2020/RubyonRails-blog/assets/55840999/9230daa8-3b08-4a7e-8266-e4672360ca5a)

- [ ] Same errors for 52 and 60 line tests.

By addressing this test case, you'll improve the overall reliability of your tests. Keep up the good work! 🚀🛠️